handler/engine/src/main/java/io/knotx/fragments/engine/graph/CompositeNode.java

```diff
@@ -19,7 +19,7 @@
 
 public interface CompositeNode extends Node {
 
-  List<Node> getNodes();
+  <T extends Node> List<T> getNodes();
```

Comment:
Won't it always be the same type per class implementing the interface? If so, I think it'd be better to make the whole interface generic instead of only this method